packages/twenty-front/src/modules/object-record/record-table/record-table-cell/hooks/useUpsertRecord.ts

```diff
@@ -18,6 +19,11 @@ export const useUpsertRecord = ({
     objectNameSingular,
   });
 
+  const { labelIdentifierFieldMetadataItem } =
```

Comment:
this is introducing a re-render but this is fine:
- useUpsertRecord goal is to provide a function that can be called at very specific moment. We don't need it to be reactive, that's why we use recoil callback. It would be better from a coding pattern to get the labelIdentifierFieldMetadataItem from the snapshot within the recoil callback
- however here labelIdentifierFieldMetadataItem is part of metadata. Metadata changes will trigger a re-render of the whole app anyway so this is acceptable

Reply:
I've made the change you recommended, using snapshot (even though it doesn't impact performance - you're right, best to keep the best practise)